server/modules/survey/manager/surveyManager.js

```diff
@@ -185,16 +190,13 @@ const updateSurveyProps = async (user, surveyId, props, client = db) =>
     }
   })
 
-const publishSurveyProps = async (surveyId, langsDeleted, client = db) => {
-  await SurveyRepository.publishSurveyProps(surveyId, client)
-
-  for (const langDeleted of langsDeleted) {
-    await Promise.all([
-      SurveyRepository.deleteSurveyLabel(surveyId, langDeleted, client),
-      SurveyRepository.deleteSurveyDescription(surveyId, langDeleted, client),
-    ])
-  }
-}
+const publishSurveyProps = async (surveyId, langsDeleted, client = db) =>
+  client.tx(async t => {
```

Comment:
should we always use await?

Reply:
it works the same (because the return value is always a Promise) but good catch